src/ui/components/Library/Team/View/Recordings/Header/BatchActionDropdown.tsx

```diff
@@ -114,11 +115,16 @@ export default function BatchActionDropdown({
       <span>{`${selectedIds.length} item${selectedIds.length > 1 ? "s" : ""} selected`}</span>
     </span>
   );
+
+  const selectedRecordings = selectedIds.map(id => recordings.find(r => r.id === id));
   // Disable moving the selected recordings to the library if the user is not the author of
   // all the selected recordings.
-  const enableLibrary = selectedIds
-    .map(id => recordings.find(r => r.id === id))
-    .every(recording => userId === recording?.user?.id);
+  const enableLibrary = selectedRecordings.every(recording => userId === recording?.user?.id);
+  const testSuiteReplayFlags = selectedRecordings
+    .filter(r => r != null)
+    .map(r => isTestSuiteReplay(r!));
+  const allTestSuiteReplays = testSuiteReplayFlags.every(isTest => isTest === true);
+  const enableMove = allTestSuiteReplays || testSuiteReplayFlags.every(isTest => isTest === false);
```

Comment:
nit: I think we can reduce the amount of looping/mapping and make this more efficient (but feel free to ignore)

```js
let allRecordingsOwnedByCurrentUser = true;
let allRecordingsTestSuites = true;
let noRecordingsTestSuites = true;

selectedIds.forEach((id) => {
  const recording = recordings.find((recording) => recording.id === id);
  assert(recording); // <- This seems worth doing regardless

  if (userId !== recording.user?.id) {
    allRecordingsOwnedByCurrentUser = false;
  }

  if (isTestSuiteReplay(recording)) {
    noRecordingsTestSuites = false;
  } else {
    allRecordingsTestSuites = false;
  }
});

const enableLibrary = allRecordingsOwnedByCurrentUser;
const enableMove = allRecordingsTestSuites || noRecordingsTestSuites;
```